packages/instantsearch.js/src/lib/utils/__tests__/createSendEventForHits-test.ts

```diff
@@ -524,6 +528,29 @@ describe('createSendEventForHits', () => {
     });
   });
 
+  it('sends event with an up-to-date index name', () => {
+    const { sendEvent, instantSearchInstance, helper, hits } =
+      createTestEnvironment({
+        index: 'index1',
+      });
+
+    sendEvent('view', hits, 'Products Viewed');
+    expect(instantSearchInstance.sendEventToInsights).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        payload: expect.objectContaining({ index: 'index1' }),
+      })
+    );
+
+    helper.setIndex('index2');
```

Comment:
This is fine, but a sortBy could be interesting too (or just a comment saying sortBy does this behind the hood), in case it doesn’t exactly call setIndex in a future

Reply:
I'll opt for a comment here if it's okay.